src/test/java/org/reaktivity/nukleus/maven/plugin/internal/generated/ListWithEnumAndVariantWithDefaultFW.java

```diff
@@ -0,0 +1,279 @@
+/**
+ * Copyright 2016-2020 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.nukleus.maven.plugin.internal.generated;
+
+import java.text.MessageFormat;
+
+import org.agrona.BitUtil;
+import org.agrona.DirectBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.reaktivity.reaktor.internal.test.types.Flyweight;
+import org.reaktivity.reaktor.internal.test.types.ListFW;
+import org.reaktivity.reaktor.internal.test.types.inner.VariantOfListFW;
+
+public final class ListWithEnumAndVariantWithDefaultFW extends ListFW
+{
+    private static final int INDEX_FIELD1 = 0;
+
+    private static final long MASK_FIELD1 = 1 << INDEX_FIELD1;
+
+    private static final int INDEX_FIELD2 = 1;
+
+    private static final long MASK_FIELD2 = 1 << INDEX_FIELD2;
+
+    private static final EnumWithVariantOfUint64 DEFAULT_VALUE_FIELD1 = EnumWithVariantOfUint64.TYPE3;
+
+    private static final EnumWithUint8 DEFAULT_VALUE_FIELD2 = EnumWithUint8.ICHI;
```

Comment:
In the `idl`, `field2` is a `variant of uint8` which happens to switch on an enum.

Doesn't that imply the default would be a `uint8` not an `enum` since the default value seems like it should remain unchanged, even if the switch type was different?

In the `list` case we omit fields that are defaulted, but if we consider what would happen if the default were encoded explicitly, then it will help to determine the correct type for the default.

I think we would call set on the builder for the `variant of uint8` type with a default `uint8` value, then the builder would internally determine the switch value and encode accordingly.

This implies that in the `idl` the default value after the `=` for `field2` would be a `uint8` value, not an `enum`.